### ISSUE TITLE
[regression] installing xcocoapods instead of cocoapods

### ISSUE THREAD

**Issue by SMillerDev** [fastlane/fastlane#18517]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Information
- Breaking version: 2.180.0
- Last working version: 2.179.0

### Regression Description
Both versions installed through brew and the fastlane update accompanied a ruby update.
Using the cocoapods plugin suddendly started pulling in [xcocoapods](https://rubygems.org/gems/xcocoapods) instead of cocoapods. It's weird to me that fastlane vendors gems that are available as brew installed formulae, but that is not part of the bug report. 

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>[13:52:46]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[13:52:47]: ----------------------------------------
[13:52:47]: --- Step: Verifying fastlane version ---
[13:52:47]: ----------------------------------------
[13:52:47]: Your fastlane version 2.180.0 matches the minimum requirement of 2.64.0  ✅
[13:52:47]: ------------------------------
[13:52:47]: --- Step: default_platform ---
[13:52:47]: ------------------------------
[13:52:47]: Driving the lane 'ios pods' 🚀
[13:52:47]: -----------------------
[13:52:47]: --- Step: cocoapods ---
[13:52:47]: -----------------------
[13:52:47]: Installing Ruby gem 'cocoapods'...
[13:52:48]: Found gem "xcocoapods" instead of the required name "cocoapods"
[13:52:50]: Successfully installed 'cocoapods'
[13:52:50]: $ pod install
[13:52:51]: ▸ [!] The version of CocoaPods used to generate the lockfile (1.10.1) is higher than the version of the current executable (1.5.3). Incompatibility issues may arise.
[13:52:51]: ▸ Analyzing dependencies</pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 11.2.2                                                  |
| Ruby                        | 2.7.3                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.31.1                                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.180.0/libexec/bin/fastlane |
| Host                        | macOS 11.2.2 (20D80)                                    |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.7/2.7.3/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1k  25 Mar 2021                             |
| Is contained                | false                                                   |
| Is homebrew                 | true                                                    |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 12.4                                                    |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     |             |   |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.180.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                           | Version      |
| ----------------------------- | ------------ |
| did_you_mean                  | 1.4.0        |
| slack-notifier                | 2.3.2        |
| atomos                        | 0.1.3        |
| CFPropertyList                | 3.0.3        |
| claide                        | 1.0.3        |
| colored2                      | 3.1.2        |
| nanaimo                       | 0.3.0        |
| xcodeproj                     | 1.19.0       |
| rouge                         | 2.0.7        |
| xcpretty                      | 0.3.0        |
| terminal-notifier             | 2.0.0        |
| unicode-display_width         | 1.7.0        |
| terminal-table                | 1.8.0        |
| plist                         | 3.6.0        |
| public_suffix                 | 4.0.6        |
| addressable                   | 2.7.0        |
| multipart-post                | 2.0.0        |
| word_wrap                     | 1.0.0        |
| tty-screen                    | 0.8.1        |
| tty-cursor                    | 0.7.1        |
| tty-spinner                   | 0.9.3        |
| artifactory                   | 3.0.15       |
| babosa                        | 1.0.4        |
| colored                       | 1.2          |
| highline                      | 1.7.10       |
| commander-fastlane            | 4.4.6        |
| excon                         | 0.79.0       |
| faraday-net_http              | 1.0.1        |
| ruby2_keywords                | 0.0.4        |
| faraday                       | 1.3.0        |
| unf_ext                       | 0.0.7.7      |
| unf                           | 0.1.4        |
| domain_name                   | 0.5.20190701 |
| http-cookie                   | 1.0.3        |
| faraday-cookie_jar            | 0.0.7        |
| faraday_middleware            | 1.0.0        |
| fastimage                     | 2.2.3        |
| gh_inspector                  | 1.1.3        |
| json                          | 2.3.0        |
| mini_magick                   | 4.11.0       |
| naturally                     | 2.2.1        |
| rubyzip                       | 2.3.0        |
| security                      | 0.1.3        |
| xcpretty-travis-formatter     | 1.0.1        |
| dotenv                        | 2.7.6        |
| bundler                       | 2.2.15       |
| simctl                        | 1.6.8        |
| jwt                           | 2.2.2        |
| uber                          | 0.1.0        |
| declarative                   | 0.0.20       |
| declarative-option            | 0.1.0        |
| representable                 | 3.0.4        |
| retriable                     | 3.1.2        |
| mini_mime                     | 1.1.0        |
| multi_json                    | 1.15.0       |
| signet                        | 0.15.0       |
| memoist                       | 0.16.2       |
| os                            | 1.1.1        |
| googleauth                    | 0.16.1       |
| httpclient                    | 2.8.3        |
| google-api-client             | 0.38.0       |
| google-cloud-env              | 1.5.0        |
| google-cloud-errors           | 1.1.0        |
| google-cloud-core             | 1.6.0        |
| rexml                         | 3.2.3.1      |
| webrick                       | 1.6.1        |
| google-apis-core              | 0.3.0        |
| google-apis-iamcredentials_v1 | 0.3.0        |
| google-apis-storage_v1        | 0.3.0        |
| rake                          | 13.0.3       |
| digest-crc                    | 0.6.3        |
| google-cloud-storage          | 1.31.0       |
| emoji_regex                   | 3.2.2        |
| jmespath                      | 1.4.0        |
| aws-partitions                | 1.442.0      |
| aws-eventstream               | 1.1.1        |
| aws-sigv4                     | 1.2.3        |
| aws-sdk-core                  | 3.113.1      |
| aws-sdk-kms                   | 1.43.0       |
| aws-sdk-s3                    | 1.93.0       |
| uri                           | 0.10.0       |
| forwardable                   | 1.3.1        |
| logger                        | 1.4.2        |
| cgi                           | 0.1.0        |
| date                          | 3.0.0        |
| timeout                       | 0.1.0        |
| stringio                      | 0.1.0        |
| zlib                          | 1.1.0        |
| ipaddr                        | 1.2.2        |
| openssl                       | 2.1.2        |
| ostruct                       | 0.2.0        |
| strscan                       | 1.0.3        |
| delegate                      | 0.1.0        |
| fileutils                     | 1.4.1        |
| io-console                    | 0.5.6        |
| open3                         | 0.1.0        |
| singleton                     | 0.1.0        |
| yaml                          | 0.1.0        |
| psych                         | 3.1.0        |
| mutex_m                       | 0.1.0        |
</details>

**Comment by joshdholtz** [comment]
@SMillerDev This is super weird! I’m looking into this issue right away

**Comment by SMillerDev** [comment]
Can I suggest checking for the `pod` command, rather than just for the gem? I can't be the only person to be surprised by this flow 😅

**Comment by joshdholtz** [comment]
@SMillerDev 

This seems to be an issue with `Gem::SpecFetcher.Fetcher` somehow 🤔 We are essentialy calling `fetcher.suggest_gems_from_name('cocoapods')` and on my end its returning only `cocoapods` (like it should be).

I’m very confused on why its returning `xcocoapods` for you 😬 

We _are_ grabbing the first result without checking it so I think I’m going to add some filtering to make sure that it finds exactly what it’s looking for 🤷‍♂️

**Comment by joshdholtz** [comment]
Okay, I got a fix in for this but I’m still SUPER confused how `xcocoapods` was even suggested 😱 

There is something crazy with the homebrew version of Ruby that returns...

```ruby
irb(main):001:0> Gem::SpecFetcher.fetcher.suggest_gems_from_name("cocoapods")
=> ["xcocoapods", "cocoapods-BDTransform", "cocoapods-Framework", "cocoapods-NameMappedModule", "cocoapods-RepoPush"]
```

For some reason `cocoapods` isn’t even an option????

My `ruby-install` version of 2.7 returns 

```rb

irb(main):001:0> Gem::SpecFetcher.fetcher.suggest_gems_from_name("cocoapods")
=> ["cocoapods"]
```

@SMillerDev Any idea why the homebrew version of Ruby 2.7 would return something completely different??? 🤯

**Comment by SMillerDev** [comment]
Did you update to `2.7.3` yet? Maybe that is it?

**Comment by joshdholtz** [comment]
Yup yup, I’m currently running `2.7.3`.  I jumped into an `irb` in Ruby `2.5.8` from homebrew and that returns the expected `cocoapods`. I might have to lock the `fastlane` formula to Ruby 2.6 for now 🤷‍♂️ Seems there is an issue with Ruby 2.7 😬

**Comment by joshdholtz** [comment]
My `ruby-install` version was `2.7.1`. I’m going to install `2.7.3` now to see if that breaks things 😛

**Comment by joshdholtz** [comment]
Confirmed that its an issue specifically with Ruby `2.7.3` from homebrew 😱 

Ruby `2.7.3` installed through `ruby-install` returns the expected `cocoapods` 🤔

**Comment by joshdholtz** [comment]
@SMillerDev Last ping here! 😇 

Do you know if there is a for me to update the [fastlane formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/fastlane.rb) to use Ruby 2.6? I can tell it to depend on 2.6 but it still uses my 2.7 if I have it installed 🤷‍♂️

**Comment by SMillerDev** [comment]
> Do you know if there is a for me to update the fastlane formula to use Ruby 2.6? I can tell it to depend on 2.6 but it still uses my 2.7 if I have it installed 🤷‍♂️

You'll need to do `brew reinstall -s fastlane` to make sure it takes local changes.

**Comment by joshdholtz** [comment]
I forgot the `-s` 🤦‍♂️  But that seemed to fix it for me! I’m going to submit a homebrew PR to lock in at Ruby 2.6 (for now) until I come up with another solution

**Comment by joshdholtz** [comment]
@SMillerDev I got an actual fix up for this now 😇 in PR #18520 ☝️
